audit_log/headers.py

```diff
@@ -65,14 +66,14 @@ def parse_spiffe(xfcc_header: str) -> ParsedSPIFFE:
 
 
 def get_principal_from_headers(
-    headers: dict[str, str],
+    headers: typing.Mapping[str, str],
```

Comment:
typing.Mapping is deprecated since 3.9: https://docs.python.org/3/library/typing.html#typing.Mapping